### DISCUSSION TITLE
Working Authelia - Configuration uncertainity

### DISCUSSION THREAD

**Discussion by mamema** [authelia/authelia#5987] [Q&A] [answered]
Hi,

i have a dockerized firefox web browser running. 
Secured with authelia. Everything is fine.
But because of Authelia security, when i launch a DNS name which is actually a local running service (from point of view of the published web browser)within the published browser, the known ip is of course the internet facing ip and not local.
So within this published local web browser, i get a authelia 2FA request (again) which doesn't work at the end. 
How can this be solved

1.) working
Internet -> DNS service ABC - > Authelia -> Access Granted to DNS service ABC (working)

2.) not working
Internet - DNS service for Firefox Browser -> Authelia -> Access Granted to DNS service Firefox -> using DNS service ABC (not working)

**Comment by james-d-elliott** [comment, marked as answer]
Not sure I have a clear picture of the issue. However if you have to authenticate all over again then the firefox container is not passing the cookie. If you have to elevate to 2FA (i.e. were not 2FA'd before) you need to configure your DNS or NAT so that the Remote IP is correct. You can either configure the DNS with a split horizon local DNS server or via the docker DNS configuration options. Basically you want to ensure that the Firefox container resolves the endpoint to a local address not a global one. This would probably be a good idea regardless.

After you fix the DNS issue if that doesn't solve the entire problem your only solutions are that firefox passes the session cookie down the line in the request chain, or you specifically create a bypass for that container using the Remote IP (since it should already be authenticated). It should be noted if you do this you must give the firefox container a static IP.
